### PR TITLE
feat: use Titillium Web font for UI

### LINKED ISSUE #768: Use Titillium Web font in UI
We wanted this on brand tagline sections; if adding there can add to whole extension.
Can get it from [FontSource](https://fontsource.org/fonts/titillium-web/) to avoid dependency on Google Fonts

Ref: [Slack #web-monetization-extension](https://interledgerfoundation.slack.com/archives/C06CVJRT5N3/p1733485547075049?thread_ts=1733412623.397759&cid=C06CVJRT5N3)
Older issue where we decided we don't need to use this font: https://github.com/interledger/web-monetization-extension/issues/374

### PR DESCRIPTION
**Pull request by DarianM**
## Context

Closes #768

### CODE REVIEW

File: src/manifest.json [hunk 1]
```diff
@@ -31,6 +31,13 @@
     {
       "resources": ["polyfill/*", "pages/progress-connect/*"],
       "matches": ["<all_urls>"]
+    },
+    {
+      "resources": [
+        "assets/fonts/titillium-web/*.woff2",
+        "assets/fonts/titillium-web/*.woff"
```
Comment (sidvishnoi) [inline src/manifest.json:38]: Can use something like `assets/fonts/**/*` maybe?

File: esbuild/plugins.ts [hunk 2]
```diff
@@ -67,6 +67,10 @@ export const getPlugins = ({
           from: toPosix(path.join(SRC_DIR, 'assets', '**', '*')),
           to: toPosix(path.join(outDir, 'assets')),
         },
+        {
+          from: ['./node_modules/@fontsource/titillium-web/files/*'],
```
Comment (sidvishnoi) [inline esbuild/plugins.ts:71]: Might need to update copy plugin to only copy required files (maybe add a filter function).

Comment (raducristianpopa) [reply esbuild/plugins.ts:71]: `@fontsource/titillium-web` ships all font variants in the `files` directory. We should only copy the variants that we are using. This way we are not going to increase the extension bundle size with assets that are not used at all.

File: esbuild/dev.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export const getDevOptions = ({
     sourcemap: 'linked',
     metafile: false,
     minify: false,
+    loader: { '.woff2': 'file' },
```
Comment (DarianM) [inline esbuild/dev.ts:18]: need to get rid of this .woff2 loader

File: src/pages/popup/index.css [hunk 4]
```diff
@@ -1,3 +1,10 @@
+@font-face {
+  font-family: 'Titillium Web';
+  src: url('../../assets/fonts/titillium-web/titillium-web-latin-400-normal.woff2')
+    format('woff2');
+  font-style: normal;
+  font-weight: 400;
```
Comment (sidvishnoi) [inline src/pages/popup/index.css:6]: Will need a bold weight also.

File: src/pages/progress-connect/index.css [hunk 5]
```diff
@@ -1,3 +1,10 @@
+@font-face {
+  font-family: 'Titillium Web';
+  src: url('../../assets/fonts/titillium-web/titillium-web-latin-400-normal.woff2')
+    format('woff2');
+  font-style: normal;
+  font-weight: 400;
+}
```
Comment (raducristianpopa) [inline src/pages/progress-connect/index.css:7]: Is this not working if we add this below the `@tailwind` parts?

File: src/pages/popup/index.css [hunk 6]
```diff
@@ -37,5 +44,6 @@
 
   body {
     @apply bg-white text-base text-medium;
+    font-family: 'Titillium Web', sans-serif;
```
Comment (raducristianpopa) [inline src/pages/popup/index.css:47]:
```suggestion
    @apply bg-white text-base text-medium font-titillium;
```

File: src/pages/progress-connect/index.css (hunk 5 above)
Comment (DarianM) [reply src/pages/progress-connect/index.css:10]: sure. no problemo 👍

**Comment by DarianM** [comment]
think will have to change all `font-medium` class to `font-bold`:
before font
image


after font
image

**Comment by sidvishnoi** [comment]
Or, add the medium variant too as it's used often

File: src/pages/popup/index.css [hunk 7]
```diff
@@ -1,6 +1,32 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+@font-face {
```
Comment (sidvishnoi) [inline src/pages/popup/index.css:4]: Do we want these in `@layer base`? 
https://tailwindcss.com/docs/font-family#customizing-your-theme

**Review by sidvishnoi — APPROVED** [review]

File: src/pages/popup/index.css
Comment (raducristianpopa) [inline src/pages/popup/index.css:1]: Curious if we can have only one CSS file. Will explore this later on.

**Review by raducristianpopa — APPROVED** [review]

File: src/pages/popup/index.css
Comment (sidvishnoi) [reply src/pages/popup/index.css:1]: Relevant: https://github.com/interledger/web-monetization-extension/pull/761#discussion_r1871176482

**Pull request merged**